FixedArray.swift

```diff
@@ -0,0 +1,70 @@
+//
+//  JTFixedArray.swift
+//  
+//  A fixed length array type in Swift.
+//
+//  Created by Ben on 26/05/2016.
+
+import Foundation
+
+struct JTFixedArray<T: Comparable> {
+	private let maxSize: Int
+	private let defaultValue: T!
+	private var array: [T]
+	private (set) var count = 0
+	
+	init(maxSize: Int, defaultValue: T) {
+		self.maxSize = maxSize
+		self.defaultValue = defaultValue
+		self.array = [T](count: maxSize, repeatedValue: defaultValue)
+	}
+	
+	subscript(index: Int) -> T {
+		get {
+			assert(index < maxSize)
+			assert(index >= 0)
+			return array[index]
+		}
+		
+		set(newValue) {
+			assert(index >= 0)
+			assert(count < maxSize)
+			count += 1
+			array[index] = newValue
+		}
+	}
+	
+	mutating func append(newElement: T) {
+		assert(count < maxSize)
+		count += 1
+		array[count] = newElement
+	}
```

Comment:
It has X defaultValue elements. If you read the `init` you can see that I give it a `defaultValue` which can be `0`, `""` etc. So the array will be `["", "", "", ""]` if you give it a `maxSize` of 4.

The count starts at 0. So, given the `assert` is met, I increment the count and assign that index of the array to the value I want it it to be. That's why `T` is the `newElement` argument.